### PR TITLE
Add 32-bit build

### PR DESCRIPTION
**Pull request by chfast**
Depends on #744.

### CODE REVIEW

**Comment by axic** [comment]
@chfast Can we get this merged by disabling FP tests on 32-bit, for now?

**Comment by chfast** [comment]
Yes.

File: cmake/ProjectWabt.cmake [hunk 1]
```diff
@@ -53,7 +64,7 @@ ExternalProject_Add(wabt
     ${build_command}
     INSTALL_COMMAND ""
     BUILD_BYPRODUCTS ${wabt_library}
-)
+    )
```
Comment (chfast) [inline cmake/ProjectWabt.cmake:67]: To be fixed.

File: circle.yml [hunk 2]
```diff
@@ -482,6 +482,19 @@ jobs:
...
+    docker:
+      - image: ethereum/cpp-build-env:15-gcc-10-multilib
+    steps:
+      - install_testfloat
+      - checkout
+      - build:
+          build_type: Debug
+          cmake_options: -DCMAKE_TOOLCHAIN_FILE=~/project/cmake/toolchains/32bit.cmake
+      - test
+      # TODO: Disabled, because fixes to NaN payload handling in the tool are needed.
```
Comment (axic) [inline circle.yml:495]: As long as you are 100% sure it is only a tooling fix (and no fix/change needed in the interpreter), I'm fine merging this.

Comment (chfast) [reply circle.yml:495]: I'm not sure, but this is the first thing to fix.

File: test/unittests/execute_floating_point_test.cpp [hunk 3]
```diff
@@ -1145,29 +1148,26 @@ TEST(execute_floating_point, f32_store)
...
-        {-1.0f, "cc000080bfcc"_bytes},
-        {std::nextafter(1.0f, 0.0f), "ccffff7f3fcc"_bytes},
-        {std::nextafter(-1.0f, 0.0f), "ccffff7fbfcc"_bytes},
-        {FP32::nan(FP32::canon), "cc0000c07fcc"_bytes},
-        {-FP32::nan(FP32::canon), "cc0000c0ffcc"_bytes},
-        {FP32::nan(FP32::canon + 1), "cc0100c07fcc"_bytes},
-        {-FP32::nan(FP32::canon + 1), "cc0100c0ffcc"_bytes},
-        {FP32::nan(1), "cc0100807fcc"_bytes},
-        {-FP32::nan(1), "cc010080ffcc"_bytes},
+    const std::tuple<float, bytes> test_cases[]
```
Comment (gumb0) [inline test/unittests/execute_floating_point_test.cpp:1151]: maybe `clang-format off` here?

File: test/unittests/execute_floating_point_test.cpp [hunk 4]
```diff
@@ -1360,6 +1372,10 @@ TEST(execute_floating_point, f64_add_off_by_one)
     _FPU_SETCW(fpu_cw);
     */
 
+#if defined(__i386__) && !defined(__SSE_MATH__)
```
Comment (gumb0) [inline test/unittests/execute_floating_point_test.cpp:1375]: why not at the top of the test?

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:1375]: To execute as much as possible. Can be at the top too.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:1375]: On top it would be more clear that long comment there explains the reason for skipping IMHO.

**Review by gumb0 — APPROVED** [review]

File: test/unittests/execute_floating_point_test.cpp (hunk 4 above)
Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:1375]: I agree. Will do it.

**Pull request merged**